Common/Chairloader/IChairServiceProvider.h

```diff
@@ -0,0 +1,21 @@
+//
+// Created by theli on 4/27/2025.
+//
+
+#ifndef ISERVICEPROVIDER_H
+#define ISERVICEPROVIDER_H
+#include <memory>
+#include <typeindex>
+
+struct IChairServiceProvider {
+    virtual ~IChairServiceProvider() = default;
+
+    virtual std::shared_ptr<void> GetService(std::type_index serviceType) = 0;
```

Comment:
Since services live as long as ServiceProvider does, shared pointer can be replaced with a plain non-owning pointer. In that case ServiceProvider will be the owner of all service instances.